Packs/Cryptosim/Integrations/Cryptosim/Cryptosim.py

```diff
@@ -0,0 +1,234 @@
+from datetime import datetime, timedelta
+from CommonServerPython import *
+from CommonServerUserPython import *
+import traceback
+import json
+import base64
+import requests
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()  # pylint: disable=no-member
+
+''' CONSTANTS '''
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'  # ISO8601 format with UTC, default in XSOAR
+
+
+class Client(BaseClient):
+
+    def correlation_alerts(self, start_time=None):
+        args = demisto.args()
+
+        end_time = datetime.now()
+
+        interval_time = end_time - timedelta(minutes=demisto.params().get('incidentFetchInterval', 360))
+
+        if start_time is None or start_time < interval_time:
+            start_time = interval_time
+        parameters = {
+            'startDate': args.get('startDate', start_time.isoformat()),
+            'endDate': args.get('endDate', end_time.isoformat()),
+            'showSolved': args.get('showSolved', False),
+            'crrPluginId': args.get('crrPluginId', -1),
+            'containStr': args.get('containStr', None),
+            'risk': args.get('risk', -1),
+            'srcIPPort': args.get('srcIPPort', None),
+            'destIPPort': args.get('destIPPort', None),
+            'srcPort': args.get('srcPort', None),
+            'destPort': args.get('destPort', None),
+            'riskOperatorID': args.get('riskOperatorID', "equal"),
+            "isJsonLog": True
+        }
+
+        return self._http_request("POST", url_suffix="correlationalertswithlogs",
+                                  data=json.dumps(parameters))
+
+    def correlations(self):
+        return self._http_request("GET", data={}, url_suffix="correlations")
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def correlations_command(client: Client):
+    result = client.correlations()
+
+    return CommandResults(
+        outputs_prefix='Correlations',
+        outputs_key_field='',
+        outputs=result,
+    )
+
+
+def correlation_alerts_command(client: Client):
+
+    # Call the Client function and get the raw response
+    result = client.correlation_alerts()
+
+    return CommandResults(
+        outputs_prefix='CorrelationAlerts',
+        outputs_key_field='',
+        outputs=result,
+    )
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication'
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+    :type client: ``Client``
+    :param Client: client to use
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ''
+    try:
+        if client.correlations().get('StatusCode') == 200:
```

Comment:
It's better to give some limit here to the function and not return all the correlations.

Reply:
For this version, there is no connection test end point. This is temporary function to check can xsoar connect cryptosim or not. after development, we will update this.

Reply:
Yes for now there is no limitations for correlations function. It gives all active correlations, but for next version we will have connection_test end point and limitations for commands